wodles/aws/aws_s3.py

```diff
@@ -1005,56 +994,59 @@ def _event_should_be_skipped(event_):
+                    match_start = date_match.span()[0] if date_match else None
 
-                        if not self._same_prefix(match_start, aws_account_id, aws_region):
-                            debug(f"++ Skipping file with another prefix: {bucket_file['Key']}", 3)
-                            continue
+                    if not self._same_prefix(match_start, aws_account_id, aws_region):
+                        debug(f"++ Skipping file with another prefix: {bucket_file['Key']}", 3)
+                        continue
 
-                    if self.already_processed(bucket_file['Key'], aws_account_id, aws_region):
-                        if self.reparse:
-                            debug(f"++ File previously processed, but reparse flag set: {bucket_file['Key']}", 1)
-                        else:
-                            debug(f"++ Skipping previously processed file: {bucket_file['Key']}", 1)
-                            continue
+                if self.already_processed(bucket_file['Key'], aws_account_id, aws_region, **kwargs):
+                    if self.reparse:
+                        debug(f"++ File previously processed, but reparse flag set: {bucket_file['Key']}", 1)
+                    else:
+                        debug(f"++ Skipping previously processed file: {bucket_file['Key']}", 1)
+                        continue
 
-                    debug(f"++ Found new log: {bucket_file['Key']}", 2)
-                    # Get the log file from S3 and decompress it
-                    log_json = self.get_log_file(aws_account_id, bucket_file['Key'])
-                    self.iter_events(log_json, bucket_file['Key'], aws_account_id)
-                    # Remove file from S3 Bucket
-                    if self.delete_file:
-                        debug(f"+++ Remove file from S3 Bucket:{bucket_file['Key']}", 2)
-                        self.client.delete_object(Bucket=self.bucket, Key=bucket_file['Key'])
-                    self.mark_complete(aws_account_id, aws_region, bucket_file)
+                debug(f"++ Found new log: {bucket_file['Key']}", 2)
+                # Get the log file from S3 and decompress it
+                log_json = self.get_log_file(aws_account_id, bucket_file['Key'])
+                self.iter_events(log_json, bucket_file['Key'], aws_account_id)
+                # Remove file from S3 Bucket
+                if self.delete_file:
+                    debug(f"+++ Remove file from S3 Bucket:{bucket_file['Key']}", 2)
+                    self.client.delete_object(Bucket=self.bucket, Key=bucket_file['Key'])
+                self.mark_complete(aws_account_id, aws_region, bucket_file, **kwargs)
+
+            if bucket_files['IsTruncated']:
+                new_s3_args = self.build_s3_filter_args(aws_account_id, aws_region, True, **kwargs)
+                new_s3_args['ContinuationToken'] = bucket_files['NextContinuationToken']
+                bucket_files = self.client.list_objects_v2(**new_s3_args)
 
```

Comment:
Check the script to be PEP 8 compliant regarding blank lines

Reply:
Fixed in https://github.com/wazuh/wazuh/pull/16325/commits/2aa83cb3d10527c2d77860871feca2159c8de39e